### PR TITLE
layout: grid-template getComputedStyle cleanup

### PR DESCRIPTION
**Pull request by stevennovaryo**
#### Changes

- Update Taffy version to `0.7.4`, which will fix the my bug of exposing the info of grid without any items.
- Enhancing code qualities. 
  - https://github.com/servo/servo/pull/34885#discussion_r1908546025

#### WPT Result &mdash; [Try Run](https://github.com/stevennovaryo/servo/actions/runs/12705267937)

Most the pass tests is WPT tests that uses grid without any items. Leftover subtests from the same tests are mostly related to the upcoming features and more passes are expected soon. 

WPT tests with track list that contains only `auto`(s) computed the sizes of the track incorrectly. #34925 

This also includes tests that uses keyword `none` in grid with grid items (and undefined auto-placement) and the one with incorrect value, because all of them should resolved to `auto`.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are the follow-up of #34885
- [X] There are tests for these changes

### CODE REVIEW

**Review by nicoburns — APPROVED** [review]
Looks good :) And thanks for creating the issue for the auto-tracks issue!

File: components/layout_2020/fragment_tree/box_fragment.rs [hunk 1]
```diff
@@ -89,7 +89,7 @@ pub(crate) struct BoxFragment {
 
     /// Additional information of from layout that could be used by Javascripts and devtools.
     #[serde(skip_serializing)]
-    pub detailed_layout_info: Option<DetailedLayoutInfo>,
+    pub detailed_layout_info: Option<SpecificLayoutInfo>,
```
Comment (Loirooriol) [inline components/layout_2020/fragment_tree/box_fragment.rs:92]: For consistency, I would rename this to `specific_layout_info`
